orthoinference/src/main/java/org/reactome/orthoinference/ReactionInferrer.java

```diff
@@ -179,9 +190,15 @@ private static boolean inferReactionCatalysts(GKInstance reactionInst, GKInstanc
 					}
 				}
 				
-				List<GKInstance> activeUnits = new ArrayList<GKInstance>();
-				for (GKInstance activeUnitInst : (Collection<GKInstance>) catalystInst.getAttributeValuesList(activeUnit))
+				List<GKInstance> activeUnits = new ArrayList<>();
+				Collection<GKInstance> activeUnitInstances = (Collection<GKInstance>) catalystInst.getAttributeValuesList(activeUnit);
+				logger.info("Total active unit instances: " + activeUnitInstances);
+				if (activeUnitInstances.size() > 0) {
+					logger.info("Active unit instance(s): " + activeUnitInstances);
+				}
+				for (GKInstance activeUnitInst : activeUnitInstances)
```

Comment:
The if-statement above checks that `activeUnitInstances` has elements, so this loop that depends on `activeUnitInstances`  could be moved to be within that if-statement.